# HG changeset patch
# User Michael Goffioul
# Date 1220401888 14400
#      Tue Sep 02 20:31:28 2008 -0400
# Node ID b1634dd9ebe7a21c78ed9cf2341d81a9efbef569
# Parent  9f10a7b28ae3d1785e1b4d7a65344c0f20456a1e
Make hggroup to react on [xyzac]liminclude changes in its children.

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,13 @@
+2008-09-02  Michael Goffioul  <michael.goffioul@gmail.com>
+
+	* graphics.cc (hggroup::update_axis_limits): Also reacts on
+	[xyzac]liminclude changes.
+
 2008-08-31  Michael Goffioul  <michael.goffioul@gmail.com>
 
 	* DLD-FUNCTIONS/fltk_backend.cc (fltk_backend::redraw_figure,
 	fltk_backend::print_figure): Change graphics_handle argument to
 	graphics_object.
 
 2008-08-29  David Bateman  <dbateman@free.fr>
 
diff --git a/src/graphics.cc b/src/graphics.cc
--- a/src/graphics.cc
+++ b/src/graphics.cc
@@ -3479,42 +3479,42 @@ hggroup::update_axis_limits (const std::
   Matrix kids = xproperties.get_children ();
 
   double min_val = octave_Inf;
   double max_val = -octave_Inf;
   double min_pos = octave_Inf;
 
   char update_type = 0;
 
-  if (axis_type == "xlim")
+  if (axis_type == "xlim" || axis_type == "xliminclude")
     {
       get_children_limits (min_val, max_val, min_pos, kids, 'x');
       
       update_type = 'x';
     }
-  else if (axis_type == "ylim")
+  else if (axis_type == "ylim" || axis_type == "yliminclude")
     {
       get_children_limits (min_val, max_val, min_pos, kids, 'y');
 
       update_type = 'y';
     }
-  else if (axis_type == "zlim")
+  else if (axis_type == "zlim" || axis_type == "zliminclude")
     {
       get_children_limits (min_val, max_val, min_pos, kids, 'z');
 
       update_type = 'z';
     }
-  else if (axis_type == "clim")
+  else if (axis_type == "clim" || axis_type == "climinclude")
     {
       get_children_limits (min_val, max_val, min_pos, kids, 'c');
 
       update_type = 'c';
 
     }
-  else if (axis_type == "alim")
+  else if (axis_type == "alim" || axis_type == "aliminclude")
     {
       get_children_limits (min_val, max_val, min_pos, kids, 'a');
 
       update_type = 'a';
     }
 
   Matrix limits (1, 3, 0.0);
 
